dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UserResource.java

```diff
@@ -747,6 +748,33 @@ public User updateLoggedInUserMetadata(@ApiParam(hidden = true)@Parameter(hidden
         return dbuser;
     }
 
+    @PUT
+    @Timed
+    @UnitOfWork
+    @RolesAllowed({"admin", "curator"})
+    @Path("/user/{userId}/privileges")
+    @Operation(operationId = "addUserPrivileges", description = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    @ApiOperation(value = "Updates the provided userID to admin or curator status, ADMIN or CURATOR only", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = User.class)
+    public User setUserPrivilege(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User authUser,
+                                 @ApiParam(value = "User ID", required = true) @PathParam("userId") Long userID,
+                                 @ApiParam(value = "Set privilege for a user", required = true) PrivilegeRequest privilegeRequest) {
+        User user = userDAO.findById(userID);
+        if (user == null) {
+            throw new CustomWebApplicationException("User not found", HttpStatus.SC_NOT_FOUND);
+        }
+
+        if (privilegeRequest.isAdmin() != user.getIsAdmin() && !authUser.getIsAdmin()) {
+            throw new CustomWebApplicationException("You do not have privileges to set/remove administrative rights", HttpStatus.SC_FORBIDDEN);
```

Comment:
Instead of `set/remove`, I suggest `modify`

Reply:
Done.